### ISSUE TITLE
When map is destroyed with polyline added twice to it, a NPE is thrown at PolyOverlayWithIW.onDetach()

### ISSUE THREAD

**Issue by farmazon3000** [osmdroid/osmdroid#1549]
I'm using ```org.osmdroid:osmdroid-android:6.1.6```.
I have a MapView with Polyline added - I added the same instance of polyline more than once:
```
map.overlayManager.add(polyline) //add 1st time
map.overlayManager.add(polyline) //add 2nd time
```
 Then I go back from Activity with map via pressing back button.
Activity reports onPause, onStop, onDestroy.
Then NPE is reported:
```
AndroidRuntime: Shutting down VM
AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.testdrugi, PID: 25077
    java.lang.NullPointerException: Attempt to invoke virtual method 'void org.osmdroid.views.overlay.LinearRing.clear()' on a null object reference
        at org.osmdroid.views.overlay.PolyOverlayWithIW.onDetach(PolyOverlayWithIW.java:351)
        at org.osmdroid.views.overlay.Polyline.onDetach(Polyline.java:135)
        at org.osmdroid.views.overlay.DefaultOverlayManager.onDetach(DefaultOverlayManager.java:195)
        at org.osmdroid.views.MapView.onDetach(MapView.java:1050)
        at org.osmdroid.views.MapView.onDetachedFromWindow(MapView.java:1255)
        at android.view.View.dispatchDetachedFromWindow(View.java:18408)
        at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3779)
        at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3771)
        at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3771)
        at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3771)
        at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3771)
        at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3771)
        at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3771)
        at android.view.ViewRootImpl.dispatchDetachedFromWindow(ViewRootImpl.java:3849)
        at android.view.ViewRootImpl.doDie(ViewRootImpl.java:6805)
        at android.view.ViewRootImpl.die(ViewRootImpl.java:6782)
        at android.view.WindowManagerGlobal.removeViewLocked(WindowManagerGlobal.java:458)
        at android.view.WindowManagerGlobal.removeView(WindowManagerGlobal.java:396)
        at android.view.WindowManagerImpl.removeViewImmediate(WindowManagerImpl.java:124)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4735)
        at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:39)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1977)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6923)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870)
```

**Comment by farmazon3000** [comment]
Why I'm using "don't keep activities"? I wanted to check how added polylines survives Activity recreation because I couldn't find it in docs.

**Comment by farmazon3000** [comment]
OK, so far I've found only hidden secret that [onPause and onResume should be called from activity](https://github.com/osmdroid/osmdroid/blob/master/osmdroid-android/src/main/java/org/osmdroid/views/MapView.java#L55). Anyway - adding them did not help me.

**Comment by farmazon3000** [comment]
mOverlay is null [here](https://github.com/osmdroid/osmdroid/blob/master/osmdroid-android/src/main/java/org/osmdroid/views/overlay/PolyOverlayWithIW.java#L351), why?

**Comment by farmazon3000** [comment]
Why when "don't keep activities" is enabled, then Android calls MapView's onDetach twice? Because this is what happens!

**Comment by farmazon3000** [comment]
Of course adding it twice was a bug on my side, but still - as map allows to doing so, it should also gently handle it or refuse to add twice.

**Comment by monsieurtanuki** [comment]
@farmazon3000 Thank you for the tip.
Closed: fixed by #1553.

**Comment by diegoseresco** [comment]
Thanks!! It helped me a lot!
